packages/server/lib/controllers/oauth.controller.ts

```diff
@@ -989,6 +1018,40 @@ class OAuthController {
                 };
             }
 
+            if (connectionConfig['oauth_client_id'] && connectionConfig['oauth_client_secret']) {
+                parsedRawCredentials = {
+                    ...parsedRawCredentials,
+                    config_override: {
+                        client_id: connectionConfig['oauth_client_id'],
+                        client_secret: connectionConfig['oauth_client_secret']
+                    }
+                };
+
+                connectionConfig = Object.keys(session.connectionConfig).reduce((acc: Record<string, string>, key: string) => {
+                    if (key !== 'oauth_client_id' && key !== 'oauth_client_secret') {
+                        acc[key] = connectionConfig[key] as string;
+                    }
+                    return acc;
+                }, {});
```

Comment:
maybe it's just me but I have hard time reading reducer, I don't really understand what's going on

Reply:
I don't mind `reduce` personally but it would be way more readable in typescript if the initial value was the first parameter instead of the last one after the function. Sometimes I have a hard time understanding language creators/maintainers.

Reply:
I would have rather just done
```
delete connectionConfig.oauth_client_id;
delete connectionConfig.oauth_client_secret;
```
but eslint was yelling at me.

Reply:
ah yes I see, I think it's fair to disable some eslint errors when we don't find them relevant. 
And if it's often the case, it's always good to revisit the list of activated rules 👌🏻